File: markdowner-core/src/test/java/org/jesperancinha/parser/markdowner/MergeParserHelperTest.java
package org.jesperancinha.parser.markdowner;

import org.jesperancinha.parser.markdowner.model.Paragraphs;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MergeParserHelperTest {

    @Test
    public void mergeDocumentWithFooterTemplate() {
        final String testReadme = "Readme text!";
        final Paragraphs.ParagraphsBuilder paragraphsBuilder = new Paragraphs.ParagraphsBuilder();
        paragraphsBuilder.withTagParagraph("## tag", "This is a test paragraph");
        paragraphsBuilder.withTagParagraph("## ta", "This is a test2 paragraph");
        final Paragraphs testParagraphs = paragraphsBuilder.build();

        final String result = MergeParserHelper.mergeDocumentWithFooterTemplate(testReadme, testParagraphs);

        assertThat(result).isEqualTo("Readme text!\n\n## tag\nThis is a test paragraph\n\n## ta\nThis is a test2 paragraph\n");
    }

    @Test
    public void mergeDocumentWithFooterTemplateDifferentOrder() {
        final String testReadme = "Readme text!";
        final Paragraphs.ParagraphsBuilder paragraphsBuilder = new Paragraphs.ParagraphsBuilder();
        paragraphsBuilder.withTagParagraph("## ta", "This is a test2 paragraph");
        paragraphsBuilder.withTagParagraph("## tag", "This is a test paragraph");
        final Paragraphs testParagraphs = paragraphsBuilder.build();

        final String result = MergeParserHelper.mergeDocumentWithFooterTemplate(testReadme, testParagraphs);

        assertThat(result).isEqualTo("Readme text!\n\n## ta\nThis is a test2 paragraph\n\n## tag\nThis is a test paragraph\n");
    }
}

File: markdowner-core/src/test/kotlin/org/jesperancinha/parser/markdowner/helper/MergeParserHelperTest.kt
package org.jesperancinha.parser.markdowner.helper

import io.kotest.matchers.shouldBe
import org.jesperancinha.parser.markdowner.model.Paragraphs
import org.junit.jupiter.api.Test

class MergeParserHelperTest {
    @Test
    fun mergeDocumentWithFooterTemplate() {
        val testReadme = "Readme text!"
        val paragraphsBuilder = Paragraphs()
        paragraphsBuilder.withTagParagraph("## tag", "This is a test paragraph")
        paragraphsBuilder.withTagParagraph("## ta", "This is a test2 paragraph")
        val testParagraphs = paragraphsBuilder
        val result = MergeParserHelper.mergeDocumentWithFooterTemplate(testReadme, testParagraphs)
        result shouldBe "Readme text!\n\n## tag\nThis is a test paragraph\n\n## ta\nThis is a test2 paragraph\n"
    }

    @Test
    fun mergeDocumentWithFooterTemplateDifferentOrder() {
        val testReadme = "Readme text!"
        val paragraphsBuilder = Paragraphs()
        paragraphsBuilder.withTagParagraph("## ta", "This is a test2 paragraph")
        paragraphsBuilder.withTagParagraph("## tag", "This is a test paragraph")
        val testParagraphs = paragraphsBuilder
        val result = MergeParserHelper.mergeDocumentWithFooterTemplate(testReadme, testParagraphs)
        result shouldBe "Readme text!\n\n## ta\nThis is a test2 paragraph\n\n## tag\nThis is a test paragraph\n"
    }
}